practical/ProcessMining/group2/conformance_checking/src/check_conformance.py

```diff
@@ -0,0 +1,87 @@
+import sys
+import os
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.dirname(SCRIPT_DIR))
+
+
+from src.generate_footprint import (
+    FootPrintMatrix,
+)
+
+
+class ConformanceChecking:
+
+    def get_conformance_matrix(self, fpm_1, fpm_2):
+        """
+        Create a conformance matrix comparing relations of two FootPrintMatrix objects.
+
+        Parameters:
+        ----------
+        fpm_1 : FootPrintMatrix
+            The first FootPrintMatrix to compare.
+        fpm_2 : FootPrintMatrix
+            The second FootPrintMatrix to compare.
+
+        Returns:
+        -------
+        FootPrintMatrix
+            A FootPrintMatrix representing the conformance matrix.
+        """
+        dict_out = {}
+        keys_1 = set(fpm_1.relations.keys())
+        keys_2 = set(fpm_2.relations.keys())
+        all_keys = keys_1.union(keys_2)
+
+        for key in all_keys:
+            inner_dict_out = {}
+            inner_keys_1 = set(fpm_1.relations.get(key, {}).keys())
+            inner_keys_2 = set(fpm_2.relations.get(key, {}).keys())
+            all_inner_keys = inner_keys_1.union(inner_keys_2)
+
+            for inner_key in all_inner_keys:
+                inner_v1 = fpm_1.relations.get(key, {}).get(inner_key, '')
+                inner_v2 = fpm_2.relations.get(key, {}).get(inner_key, '')
+                if inner_v1 == inner_v2:
+                    inner_dict_out[inner_key] = ''
+                else:
+                    inner_dict_out[inner_key] = '{}:{}'.format(inner_v1, inner_v2)
```

Comment:
could be done by init inner_dict_out as defaultdict(str), so that you only need to handle` if inner_v1 != inner_v2`